--- a/src/anyp/Uri.cc
+++ b/src/anyp/Uri.cc
@@ -334,8 +334,9 @@
             SBufToCstring(foundHost, rawHost);
 
             if (!tok.skip(':'))
-                throw TextException("missing required :port in CONNECT target", Here());
-            foundPort = parsePort(tok);
+                foundPort = 443; // default to HTTPS port when client omits it
+            else
+                foundPort = parsePort(tok);
 
             if (!tok.remaining().isEmpty())
                 throw TextException("garbage after host:port in CONNECT target", Here());
